sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -567,74 +266,136 @@ export class TopicClient implements Client {
-    _response: HttpOperationResponse;
+// @public (undocumented)
+export interface Track2FetchOptions {
+    // (undocumented)
+    maxWaitTimeInSeconds?: number;
 }
 
-// @public
-export interface UpdateQueueResponse extends QueueDetails {
-    _response: HttpOperationResponse;
+// @public (undocumented)
+export interface Track2FetchResult<MessageT, ContextT> extends Track2CloseableThing {
+    // (undocumented)
+    context: ContextT;
+    // (undocumented)
+    iterator: AsyncIterableIterator<MessageT>;
 }
 
-// @public
-export interface UpdateRuleResponse extends RuleDetails {
-    _response: HttpOperationResponse;
+// @public (undocumented)
+export interface Track2Message extends Omit<ServiceBusMessage, "abandon" | "complete" | "defer" | "deadLetter"> {
 }
 
-// @public
-export interface UpdateSubscriptionResponse extends SubscriptionDetails {
-    _response: HttpOperationResponse;
+// @public (undocumented)
+export interface Track2MessageBatch {
+    // (undocumented)
+    tryAdd(message: Track2Message): boolean;
 }
 
-// @public
-export interface UpdateTopicResponse extends TopicDetails {
-    _response: HttpOperationResponse;
+// @public (undocumented)
+export interface Track2PeekedMessage extends ReceivedMessageInfo {
+}
+
+// @public (undocumented)
+export interface Track2PlainContext {
+}
+
+// @public (undocumented)
+export class Track2QueueConsumerClient {
+    constructor(connectionString: string, queueName: string, options?: Track2QueueConsumerClientOptions);
+    // (undocumented)
+    close(): Promise<void>;
+    // (undocumented)
+    consume(sessionId: string, mode: "PeekLock", handlers: Track2ReceiverHandlers<Track2SessionMessage, Track2SessionContext & Track2SettleableContext>): Track2CloseableThing;
+    // (undocumented)
+    consume(sessionId: string, mode: "ReceiveAndDelete", handlers: Track2ReceiverHandlers<Track2SessionMessage, Track2SessionContext>): Track2CloseableThing;
+    // (undocumented)
+    consume(mode: "PeekLock", handlers: Track2ReceiverHandlers<Track2Message, Track2SettleableContext>): Track2CloseableThing;
+    // (undocumented)
+    consume(mode: "ReceiveAndDelete", handlers: Track2ReceiverHandlers<Track2Message, Track2PlainContext>): Track2CloseableThing;
+    // (undocumented)
+    fetch(sessionId: string, mode: "PeekLock", options?: Track2FetchOptions): Track2FetchResult<Track2Message, Track2SettleableContext>;
+    // (undocumented)
+    fetch(sessionId: string, mode: "ReceiveAndDelete", options?: Track2FetchOptions): Track2FetchResult<Track2Message, Track2PlainContext>;
+    // (undocumented)
+    fetch(mode: "PeekLock", options?: Track2FetchOptions): Track2FetchResult<Track2Message, Track2SettleableContext>;
+    // (undocumented)
+    fetch(mode: "ReceiveAndDelete", options?: Track2FetchOptions): Track2FetchResult<Track2Message, Track2PlainContext>;
+    // (undocumented)
+    peekWithoutLock(sessionId: string, messageCount?: number): Promise<Track2PeekedMessage[]>;
```

Comment:
maxMessageCount since it might return less than what is set here?

Reply:
Done.

Reply:
(this method is now called receiveBatch)